### DISCUSSION TITLE
Maintaining a variable's value after stop button is pressed on Spike Prime Hub

### DISCUSSION THREAD

**Discussion by FLLCoachFinFinders** [pybricks/support#1951] [Q&A]
In our robot based on Spike Prime Hub, there are three gears that can be independently driven by a single motor. A second motor selects one of the three gears to be driven by the first motor. A python variable keeps track of which gear is currently connected to the first motor. As we run through the FLL robot game, the variable is used to change the gear being controlled as needed. This works well and has successfully been used. 

There is a problem when stop button (configured using system.set_stop_button(button)) is used. In a well behaved robot run, stop button is not used except may be at the end of the game. However, when the robot gets stuck or otherwise misbehaves, stop button may be used in the middle of the game. When stop button is used and the robot is relaunched, the variable is reset to its initial value. At that point, the variable value may no longer match the gear selected by second motor when the stop button was pressed. This means we can't relaunch the robot without manually matching the selected gear with the variable (we have created a program to do this but it takes several seconds that we can't afford to lose in a game)

I can think of two ways to solve this:

1. Run the second motor to select gear that matches the initial value whenever stop button is pressed.
    This will ensure that the variable continues to point to the selected gear when the robot is relaunched. 
    Is this possible in Pybricks? In other words, can pressing stop button launch a program that runs before stopping?

2. Create a "global" variable that maintains its value even after stop button is pressed
     In this solution, the variable is only initialized at power up. Pressing the stop button and relaunching the robot does not reset the variable. 
     Is this possible in Pybricks?

Are there other ways of achieving the main goal of knowing the gear selected when the stop button is pressed and using that info when the robot is relaunched? 

Really appreciate any ideas

**Comment by dlech** [comment]
This is what `hub.system.storage()` is for.

https://docs.pybricks.com/en/stable/hubs/primehub.html#pybricks.hubs.PrimeHub.system.storage
